include/nccl_ofi_platform.h

```diff
@@ -27,4 +27,14 @@ int platform_init(const char **provider_filter) __attribute__((weak));
  */
 int platform_config_endpoint(struct fi_info *info, struct fid_ep *ep) __attribute__((weak));
 
+/* Platform-specific hook to assign indices to rails in the multi-rail
+ * protocol of the plugin. Some providers rely on having a consistent mapping
+ * of rail indices for best performance.
+ * @param info_list: list of `num_rails` info objects
+ * @param num_rails: number of rails
+ * @param rail_indices: array of `num_rails` indices, set by this function
+ */
+void platform_get_rail_indices(struct fi_info *info_list, int num_rails,
```

Comment:
This seems rather specific to us; are we sure there's not a more general interface here?

Reply:
I can't think of one. Maybe `get_unique_guid(fi_info)`? Though that's not much different than what this function does.

Reply:
what about platform_sort_rails()?

Reply:
That would work, though I'm not sure it's meaningfully different from `platform_get_rail_indices`.

Moving the rail-sorting part to a separate PR.